spec/controllers/controllers.spec.js

```diff
@@ -0,0 +1,208 @@
+const fs = require("fs").promises;
+const path = require("path");
+
+const controller = require("../../controllers/controller");
+const helper = require("../helper/db.helper");
+const Slot = require("../../models/slot");
+
+describe("Initialize suite", function () {
+  it("should initialize the database ", async function () {
+    await helper.deleteDB();
+    
+    await controller.initializeApp();
+    
+    try {
+      const data = await fs.readFile(path.join("db", "db.json"), "utf8");
+      const lot = JSON.stringify([...Array(10).keys()].map((i) => new Slot(i)));
+      expect(data).toEqual(lot);
+    } catch (err) {
+      console.log(err);
+    }
+  });
+});
+
+describe("Park Car Suite", function () {
+  beforeEach(helper.newDB);
+
+  it("should not park a car with invalid registration number", async function () {
+    try {
+      const data = await controller.parkCar("A", Date.now());
+
+      expect(data).toEqual({ error: "Invalid Registration Number" });
+    } catch (err) {
+      console.log(err);
+    }
+  });
+
+  it("should park a car with valid registration number", async function () {
+    try {
+      const registrationNumber = "UP32EA7196";
+      const timestamp = Date.now();
+      await controller.parkCar(registrationNumber, timestamp);
+
+      const data = await fs.readFile(path.join("db", "db.json"), "utf8");
+      const lot = [...Array(10).keys()].map((i) => new Slot(i));
+      lot[0] = {
+        car: { registrationNumber: registrationNumber },
+        slotNo: 0,
+        timestamp,
+      };
+
+      expect(data).toEqual(JSON.stringify(lot));
+    } catch (err) {
+      console.log(err);
+    }
+  });
+
+  it("should not park more than 10 cars", async function () {
```

Comment:
Try to think of this spec from the perspective of - "should not park more than the size of the parking lot" and use constant for parking lot size instead of using the number `10`